types/jsdom/index.d.ts

```diff
@@ -103,7 +103,153 @@ export type ConstructorOptions = Options & {
     contentType?: string;
 };
 
-export interface DOMWindow extends Window { eval(script: string): void; }
+export interface DOMWindow extends Window {
+    eval(script: string): void;
+
+    /* node_modules/jsdom/living/index.js */
+    DOMException: typeof DOMException;
+    Attr: typeof Attr;
+    Node: typeof Node;
+    Element: typeof Element;
+    DocumentFragment: typeof DocumentFragment;
+    Document: typeof Document;
+    HTMLDocument: typeof HTMLDocument;
+    XMLDocument: typeof XMLDocument;
+    CharacterData: typeof CharacterData;
+    Text: typeof Text;
+    CDATASection: typeof CDATASection;
+    ProcessingInstruction: typeof ProcessingInstruction;
+    Comment: typeof Comment;
+    DocumentType: typeof DocumentType;
+    DOMImplementation: typeof DOMImplementation;
+    NodeList: typeof NodeList;
+    HTMLCollection: typeof HTMLCollection;
+    HTMLOptionsCollection: typeof HTMLOptionsCollection;
+    DOMStringMap: typeof DOMStringMap;
+    DOMTokenList: typeof DOMTokenList;
+    Event: typeof Event;
+    CustomEvent: typeof CustomEvent;
+    MessageEvent: typeof MessageEvent;
+    ErrorEvent: typeof ErrorEvent;
+    HashChangeEvent: typeof HashChangeEvent;
+    FocusEvent: typeof FocusEvent;
+    PopStateEvent: typeof PopStateEvent;
+    UIEvent: typeof UIEvent;
+    MouseEvent: typeof MouseEvent;
+    KeyboardEvent: typeof KeyboardEvent;
+    TouchEvent: typeof TouchEvent;
+    ProgressEvent: typeof ProgressEvent;
+    CompositionEvent: typeof CompositionEvent;
+    WheelEvent: typeof WheelEvent;
+    EventTarget: typeof EventTarget;
+    Location: typeof Location;
+    History: typeof History;
+    Blob: typeof Blob;
+    File: typeof File;
+    FileList: typeof FileList;
+    DOMParser: typeof DOMParser;
+    FormData: typeof FormData;
+    XMLHttpRequestEventTarget: XMLHttpRequestEventTarget;
+    XMLHttpRequestUpload: typeof XMLHttpRequestUpload;
+    NodeIterator: typeof NodeIterator;
+    TreeWalker: typeof TreeWalker;
+    NamedNodeMap: typeof NamedNodeMap;
+    URL: typeof URL;
+    URLSearchParams: typeof URLSearchParams;
+
+    /* node_modules/jsdom/living/register-elements.js */
+    HTMLElement: typeof HTMLElement;
+    HTMLAnchorElement: typeof HTMLAnchorElement;
+    HTMLAppletElement: typeof HTMLAppletElement;
+    HTMLAreaElement: typeof HTMLAreaElement;
+    HTMLAudioElement: typeof HTMLAudioElement;
+    HTMLBaseElement: typeof HTMLBaseElement;
+    HTMLBodyElement: typeof HTMLBodyElement;
+    HTMLBRElement: typeof HTMLBRElement;
+    HTMLButtonElement: typeof HTMLButtonElement;
+    HTMLCanvasElement: typeof HTMLCanvasElement;
+    HTMLDataElement: typeof HTMLDataElement;
+    HTMLDataListElement: typeof HTMLDataListElement;
+    // HTMLDetailsElement: typeof HTMLDetailsElement;
```

Comment:
Is there any reason why these are commented out?

Reply:
From the OP:

> Some types are not defined even globally by TypeScript. These have been commented out in the DOMWindow interface, but are still valid properties in jsdom. If someone tries to use these properties, their code will still not compile.

I kept the properties listed but commented out so we wouldn't need to dig up these properties in the future.

Reply:
Sorry I missed that; thanks!